### PR TITLE
iox-#1747 Make vector::capacity() to `static constexpr`

### LINKED ISSUE #1747: `vector::capacity()` should be `static constexpr`
## Brief feature description

The type `iox::cxx::vector` has the method `capacity()` which could be `static constexpr`. This is completely analogous to the `iox::cxx::string` type, which already has `static constexpr` method `capacity()`.

## Detailed information

I think this only has benefits, no drawbacks :)

### PR DESCRIPTION
**Pull request by JakubSosnovec**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [ ] All checks have passed (except `task-list-completed`)
1. [ ] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [ ] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes #1747

### CODE REVIEW

File: iceoryx_hoofs/test/moduletests/test_cxx_vector.cpp [hunk 1]
```diff
@@ -144,6 +145,7 @@ TEST_F(vector_test, NewlyCreatedVectorHasSizeZero)
 TEST_F(vector_test, Capacity)
 {
     ::testing::Test::RecordProperty("TEST_ID", "e0a244d5-6e01-4cbb-9f9a-ac07cad81a5c");
+    static_assert(decltype(sut)::capacity() == 10U, "Vector capacity has to match!");
```
Comment (elfenpiff) [inline iceoryx_hoofs/test/moduletests/test_cxx_vector.cpp:148]: Technically you are right but we require 100% MC/DC coverage in unit tests for our safety critical use case. Therefore I would just remove that line and rewrite the line below into 
```cxx
EXPECT_THAT(decltype(sut)::capacity(), Eq(10U));
```

**Review by elfenpiff — COMMENTED** [review]
Looks good I would merge it as soon as the clang-tidy warnings are fixed.
We use clang-tidy all over our project and you should see those warnings with a clang-tidy extension in your IDE. If you need support there please ping us.

**Review by mossmaurice — COMMENTED** [review]
@JakubSosnovec Thanks for you contribution! There are still [some failures](https://github.com/eclipse-iceoryx/iceoryx/actions/runs/3265134554/jobs/5366865252) on the `clang-tidy` job. Could you replace the calls to `capacity()` to not go through an object but directly `vector<Foo>::capacity()`?

**Comment by JakubSosnovec** [comment]
Oh, sorry guys, I didnt notice #1729  We could just close this PR, to avoid any conflicts. Or do you prefer to go with #1748 and then update #1729 ? Surely I can fix the clang-tidy problems like in #1729

**Comment by mossmaurice** [comment]
> Oh, sorry guys, I didnt notice #1729 We could just close #1747 , to avoid any conflicts. Or do you prefer to go with #1748 and then update #1729 ? Surely I can fix the clang-tidy problems like in #1729

No worries :smiley: Typically, we follow the Git development rhythm: Whoever comes first, does not have to rebase. So I don't really mind and it's up you :) But I suppose #1729 will get merged rather soon-ish.

**Comment by JakubSosnovec** [comment]
Closing as duplicate of #1729

**Pull request closed without merging**